### PR TITLE
Enhancement: Add work pool creation wizard

### PR DESCRIPTION
**Pull request by desertaxle**
## Summary
Adds work pool creation wizard to add emphasis to the available infrastructure types.

The wizard steps through basic work pool info -> infrastructure type -> infrastructure configuration. The user is shown the available infrastructure types on the infrastructure type step and an icon and description:
239041827-4a733c9b-cc07-4b5c-96e5-b0b1ae2d36dc
All information for the available infrastructure types is pulled from the prefect-collection-registry.

If the 'Prefect Agent' infrastructure type is chosen, the wizard shows the following for the infrastructure configuration step:
Screenshot 2023-05-17 at 1 46 55 PM

[Create Work Pool Wizard Demo - Watch Video](https://www.loom.com/share/a9abb2f4f3aa43258974ae930d5c8bf4)

### CODE REVIEW

**Comment by billpalombi** [comment]
This is awesome! Thanks for knocking this out!

**Review by billpalombi — APPROVED** [review]
Approved from UX perspective.

File: src/components/WorkPoolBaseJobTemplateFormSection.vue [hunk 1]
```diff
@@ -2,12 +2,11 @@
   <div>
     <h3>
       Base Job Template
-      <BetaBadge />
```
Comment (znicholasbrown) [inline src/components/WorkPoolBaseJobTemplateFormSection.vue:5]: Can you remove the import for this component as well?

File: src/components/WorkPoolCreateWizardStepInfrastructureConfiguration.vue [hunk 2]
```diff
@@ -0,0 +1,79 @@
+<template>
+  <template v-if="!typeIsPrefectAgent">
+    <p class="work-pool-create-wizard-step-infrastructure-configuration__explainer-text">
+      Below you can configure the behavior of workers in this work pool when they are executing flow runs. If you need more configuration options that the ones listed here, you can use the editor in the <b>Advanced</b> section to modify the existing configuration options.
```
Comment (znicholasbrown) [inline src/components/WorkPoolCreateWizardStepInfrastructureConfiguration.vue:4]:
```suggestion
      Below you can configure the behavior of workers in this work pool when they are executing flow runs. If you need additional configuration options, you can use the editor in the <b>Advanced</b> section to modify the existing configuration options.
```

File: src/components/WorkPoolCreateWizardStepInfrastructureConfiguration.vue [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+<script lang="ts" setup>
+  import { computed, reactive } from 'vue'
+  import { WorkPoolBaseJobTemplateFormSection } from '@/components'
+  import { useWorkspaceRoutes } from '@/compositions'
+  import { localization } from '@/localization'
+  import { WorkPoolFormValues } from '@/models/WorkPool'
+  import { WorkerBaseJobTemplate } from '@/types'
+
+  const routes = useWorkspaceRoutes()
```
Comment (znicholasbrown) [inline src/components/WorkPoolCreateWizardStepInfrastructureConfiguration.vue:32]:
```suggestion
```

File: src/components/WorkPoolCreateWizardStepInfrastructureConfiguration.vue [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      Below you can configure the behavior of workers in this work pool when they are executing flow runs. If you need more configuration options that the ones listed here, you can use the editor in the <b>Advanced</b> section to modify the existing configuration options.
+    </p>
+    <p class="work-pool-create-wizard-step-infrastructure-configuration__explainer-text">
+      If you don't need to make any changes to the default behavior, hit <b>Create</b> to create your work pool!
+    </p>
+    <WorkPoolBaseJobTemplateFormSection v-model:base-job-template="baseJobTemplate" class="work-pool-create-wizard-step-infrastructure-configuration__base-job-template-form" />
+  </template>
+  <template v-else>
+    <p class="work-pool-create-wizard-step-infrastructure-configuration__explainer-text">
+      Prefect Agents handle infrastructure configuration via infrastructure blocks attached to deployments. You can hit <b>Create</b> to create this work pool and then head over the <b>Blocks</b> tab to create an infrastructure block for your deployments.
```
Comment (znicholasbrown) [inline src/components/WorkPoolCreateWizardStepInfrastructureConfiguration.vue:13]:
```suggestion
      Prefect Agents handle infrastructure configuration via infrastructure blocks attached to deployments. You can hit <b>Create</b> to create this work pool and then head over to the <b>Blocks</b> tab to create an infrastructure block for your deployments.
```

File: src/components/WorkPoolCreateWizardStepInfrastructureConfiguration.vue [hunk 5]
```diff
@@ -0,0 +1,79 @@
+<template>
```
Comment (znicholasbrown) [inline src/components/WorkPoolCreateWizardStepInfrastructureConfiguration.vue:1]: Would be good to see this content in the localization module; if you put it there you have a convenient place to use markdown instead for the copy and then pass it to a markdown renderer here

File: src/components/WorkPoolCreateWizardStepInfrastructureConfiguration.vue [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    },
+  })
+</script>
+
+<style>
+  .work-pool-create-wizard-step-infrastructure-configuration__explainer-text { @apply
+    mb-2
+  }
+
+  .work-pool-create-wizard-step-infrastructure-configuration__base-job-template-form { @apply
```
Comment (znicholasbrown) [inline src/components/WorkPoolCreateWizardStepInfrastructureConfiguration.vue:76]:
```suggestion
.work-pool-create-wizard-step-infrastructure-configuration__base-job-template-form
```
idk if this class name is long enough 😆

File: src/components/WorkPoolCreateWizardStepInfrastructureType.vue [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+</template>
+
+<script lang="ts" setup>
+  import { useWizardStep } from '@prefecthq/prefect-design'
+  import { usePatchRef, useValidation, useValidationObserver } from '@prefecthq/vue-compositions'
+  import { computed } from 'vue'
+  import { LogoImage, BetaBadge } from '@/components'
+  import { WorkerCollectionItem } from '@/models'
+  import { WorkPoolFormValues, WorkPoolTypeSelectOption } from '@/models/WorkPool'
+  import { getProcessTypeLabel, titleCase } from '@/utilities'
```
Comment (znicholasbrown) [inline src/components/WorkPoolCreateWizardStepInfrastructureType.vue:33]:
```suggestion
  import { titleCase } from '@/utilities'
```

**Review by znicholasbrown — COMMENTED** [review]
Just a quick first pass, I won't have a chance to look at this this evening so I'll have a closer look tmr

**Review by znicholasbrown — APPROVED** [review]

**Pull request merged**